awscli/__main__.py

```diff
@@ -11,6 +11,18 @@
 # ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
 
+# Don't remove this line.  The idna encoding
+# is used by getaddrinfo when dealing with unicode hostnames,
+# and in some cases, there appears to be a race condition
+# where threads will get a LookupError on getaddrinfo() saying
+# that the encoding doesn't exist.  Using the idna encoding before
+# running any CLI code (and any threads it may create) ensures that
+# the encodings.idna is imported and registered in the codecs registry,
+# which will stop the LookupErrors from happening.
+# See: https://bugs.python.org/issue29288
+u''.encode('idna')
```

Comment:
We talked a little bit about potentially wrapping this in a try except.
Did we decide against that because effectively any networking call would attempt to use this encoding anyways and we're confident enough to not wrap it?

Otherwise lgtm, requests does something similar.

Reply:
Yeah it looks like it was added in 2003 (py23) and has a hardcoded reference in `getaddrinfo` of the socket module, so I don't think we need to wrap it.